external/avro/src/main/scala/org/apache/spark/sql/v2/avro/AvroScanBuilder.scala

```diff
@@ -41,17 +42,16 @@ class AvroScanBuilder (
       readDataSchema(),
       readPartitionSchema(),
       options,
-      pushedFilters())
+      pushedDataFilters,
+      partitionFilters,
+      dataFilters)
   }
 
-  private var _pushedFilters: Array[Filter] = Array.empty
-
-  override def pushFilters(filters: Array[Filter]): Array[Filter] = {
+  override def pushDataFilters(dataFilters: Seq[Expression]): Array[Filter] = {
```

Comment:
nit: the input parameter can be `dataFilters: Array[Filter]`, then we don't need to ask every source impl to call `translateDataFilter`